src/Functions/extractTimeZoneFromFunctionArguments.cpp

```diff
@@ -16,25 +16,22 @@ namespace ErrorCodes
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
 }
 
-
-std::string extractTimeZoneNameFromColumn(const IColumn * column, const String & column_name)
+std::string extractTimeZoneNameFromColumn(const IColumn * column, [[maybe_unused]] const String & column_name)
 {
     const ColumnConst * time_zone_column = checkAndGetColumnConst<ColumnString>(column);
 
     if (!time_zone_column)
-        throw Exception(ErrorCodes::ILLEGAL_COLUMN,
-                        "Illegal column {} of time zone argument of function, must be a constant string",
-                        column_name);
+        return "";
```

Comment:
Did you check that all callers can deal with "" as time zone? E.g. Functions/FunctionsCodingULID.cpp doesn't seem prepared. It throws an exception for empty time zone strings, not really what the user expects.

Reply:
Also suggest to carefully check all callers of `extractTimeZoneNameFromFunctionArguments()` (which you changed below)

Reply:
Yes, you are right, I've just fixed this issue